### DISCUSSION TITLE
How to set vector length for riscv v extention in Gem5

### DISCUSSION THREAD

**Discussion by xkfan** [gem5/gem5#974] [Q&A] [answered]
Hi all,

I have a question regarding the configuration of vector lengths for the RISC-V V extension in Gem5.

I know that for ARM SVE, Gem5 provides options such as 'system.sve_vl' and 'system.cpu[:].isa[:].sve_vl_se' to set the vector length. However, I am curious if there are any analogous settings available for customizing the vector length of the RISC-V V extension.

Any help is appreciated.

**Comment by rogerchang23424** [comment, marked as answer]
I've open the similar discussion here: https://github.com/orgs/gem5/discussions/838

**Reply by xkfan** [reply]
I tried the 'vlen' option, it worked.

Thanks a lot!
